common4j/src/main/com/microsoft/identity/common/java/interfaces/IPlatformComponents.java

```diff
@@ -102,6 +102,8 @@
      */
     IMultiTypeNameValueStorage getFileStore(String storeName);
 
+    INameValueStorage<String> getMultiProcessStringStore(@NonNull String storeName);
```

Comment:
Remind me..why do we need it to be shared among process?

Reply:
Because it's written by Authenticator and read in the :auth process.  This isn't actually great - I'm asking them to just go ahead and code the call to setFlights, and we can ditch this in that release.

Reply:
Also, this mode is deprecated.  Seems that multiprocess writing to shared preferences is unsafe.  There's a jetpack library called DataStorage that does this better.

Reply:
As an FYI, DataStore is not process-safe either. As for `SharedPreferences`, the reason that the `MODE_MULTI_PROCESS` became unsupported is due to the risk of corrupting the preference file if multiple processes attempt to write to the prefs file. Having a single process write to the `SharedPreferences` while another reads it using `MODE_MULTI_PROCESS` _should_ work, but you won't be notified of updates, and you have to continuously call `getSharedPreferences()` to have the underlying cache refresh.

I don't mean this to be a plug (I was just browsing through this library), but I have written a process-safe shared preference library that doesn't use ContentProviders or any other IPC Android class. It's completely file based. My approach was to use the filesystem as you would normally (files are atomic and available to all processes), and listen for changes to the file using `FileObserver`. This is the approach I took with Harmony (https://github.com/pablobaxter/Harmony). I'm more than happy to answer any questions about the approach I took with Harmony if you folks are looking for a similar approach, or questions about Harmony altogether.

Reply:
We're taking this ability as a short-term measure while we move to an IPC model for a particular piece of information in a derivative product.  As you correctly infer, only one process will actually write this file.  We could engineer a filesystem implementation, it just seems overly complex for what we're using it for.

Reply:
Yea, IPC on Android (and in general) isn't all that easy, and the constructs Android provides assumes both processes need to be running for data going back and forth, which can be impactful to the system (especially on battery). If you want a true short-term approach, I would suggest having your `SharedPrefStringNameValueStorage` call `getSharedPreferences()` instead of caching the shared prefs object. Otherwise, data won't update if you continue using the same `SharedPrefStringNameValueStorage`.